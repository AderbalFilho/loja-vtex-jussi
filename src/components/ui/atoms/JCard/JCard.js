import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import './JCard.scss';

function JCard({ children, classes, radius, shadow, size, ...rest }) {
  const className = classNames(
    'j-card',
    `j-card--rad-${radius}`,
    shadow === 'black' && 'j-card--shadow',
    `j-card--size-${size}`,
    classes
  );

  return (
    <div className={className} {...rest}>
      {children}
    </div>
  );
}

JCard.defaultProps = {
  radius: 'normal',
  shadow: 'none',
  size: 'normal',
};

JCard.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.string,
  radius: PropTypes.oneOf(['normal', 'small']),
  shadow: PropTypes.oneOf(['none', 'black']),
  size: PropTypes.oneOf(['normal', 'small']),
};

export default JCard;
