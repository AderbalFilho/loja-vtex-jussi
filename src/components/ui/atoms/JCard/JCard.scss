.j-card {
  background-color: var(--white);
  border: 1px solid var(--grey);
}

.j-card--rad-normal {
  border-radius: 8px;
}

.j-card--rad-small {
  border-radius: 4px;
}

.j-card--shadow {
  box-shadow: 0 32px 32px 0 #00000040;
}

.j-card--size-normal {
  height: 381px;
  width: 241px;
}

.j-card--size-small {
  height: 271px;
  width: 209px;
}
