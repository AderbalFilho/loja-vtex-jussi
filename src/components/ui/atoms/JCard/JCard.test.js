import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import JCard from './JCard';

describe('JCard', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<JCard>Jüssi</JCard>);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });

  describe('change radius', () => {
    it('renders normal when no radius is passed', () => {
      const { container } = render(<JCard>Jüssi</JCard>);
      const card = container.querySelector('.j-card');

      expect(card.classList).toContain('j-card--rad-normal');
    });

    it('renders small', () => {
      const { container } = render(<JCard radius="small">Jüssi</JCard>);
      const card = container.querySelector('.j-card');

      expect(card.classList).toContain('j-card--rad-small');
    });

    it('renders normal', () => {
      const { container } = render(<JCard radius="normal">Jüssi</JCard>);
      const card = container.querySelector('.j-card');

      expect(card.classList).toContain('j-card--rad-normal');
    });
  });

  it('renders shadow', () => {
    const { container } = render(<JCard shadow="black">Jüssi</JCard>);
    const card = container.querySelector('.j-card');

    expect(card.classList).toContain('j-card--shadow');
  });

  describe('change size', () => {
    it('renders normal when no size is passed', () => {
      const { container } = render(<JCard>Jüssi</JCard>);
      const card = container.querySelector('.j-card');

      expect(card.classList).toContain('j-card--size-normal');
    });

    it('renders small', () => {
      const { container } = render(<JCard size="small">Jüssi</JCard>);
      const card = container.querySelector('.j-card');

      expect(card.classList).toContain('j-card--size-small');
    });

    it('renders normal', () => {
      const { container } = render(<JCard size="normal">Jüssi</JCard>);
      const card = container.querySelector('.j-card');

      expect(card.classList).toContain('j-card--size-normal');
    });
  });

  it('renders children', () => {
    const { container } = render(<JCard>Jüssi</JCard>);
    const card = container.querySelector('.j-card');

    expect(card.textContent).toBe('Jüssi');
  });

  it('renders classes', () => {
    const { container } = render(<JCard classes="card-offer">Jüssi</JCard>);
    const card = container.querySelector('.j-card');

    expect(card.classList).toContain('card-offer');
  });
});
