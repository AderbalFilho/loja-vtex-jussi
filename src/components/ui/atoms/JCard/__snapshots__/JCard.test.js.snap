// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JCard matches snapshot 1`] = `
<div
  className="j-card j-card--rad-normal j-card--size-normal"
>
  JÃ¼ssi
</div>
`;
